x-pack/plugins/triggers_actions_ui/public/application/sections/common/components/execution_duration_chart.tsx

```diff
@@ -27,16 +27,36 @@ export interface ComponentOpts {
     average: number;
     valuesWithTimestamp: Record<string, number>;
   };
+  numberOfExecutions: number;
+  onChangeDuration: (length: number) => void;
 }
 
-const DESIRED_NUM_EXECUTION_DURATIONS = 30;
+const NUM_EXECUTIONS_OPTIONS = [120, 60, 30, 15].map((value) => ({
+  value,
+  text: i18n.translate(
+    'xpack.triggersActionsUI.sections.executionDurationChart.numberOfExecutionsOption',
+    {
+      defaultMessage: '{value} executions',
+      values: {
+        value,
+      },
+    }
+  ),
+}));
 
 export const ExecutionDurationChart: React.FunctionComponent<ComponentOpts> = ({
   executionDuration,
+  numberOfExecutions,
+  onChangeDuration,
 }: ComponentOpts) => {
   const paddedExecutionDurations = padOrTruncateDurations(
     executionDuration.valuesWithTimestamp,
-    DESIRED_NUM_EXECUTION_DURATIONS
+    numberOfExecutions
+  );
+
+  const onChange = useCallback(
+    ({ target }) => onChangeDuration(Number(target.value)),
```

Comment:
Just curious, is `Number()` different than `parseInt(target.value, 10)` in anyway?

Reply:
Yes, actually, `Number('0x64')` correctly parses to 100 but `parseInt('0x64', 10)` resolves as 0 for some reason.

Don't think we need to worry about that, though.